Add memcache support for admin bucket in helm (#3740)
grafana/mimir: operations/helm/charts/mimir-distributed/CHANGELOG.md

Entries should include a reference to the Pull Request that introduced the change.

## main / unreleased

> * [FEATURE] Add cache support for GEM's admin bucket. The cache will be enabled by default when you use
>   small.yaml, large.yaml, capped-small.yaml and capped-large.yaml helm values example. #3740
>   > **Note:** For more information, refer to the [Grafana Enterprise Metrics configuration](https://grafana.com/docs/enterprise-metrics/latest/config).

Reviewer: ```suggestion

* [FEATURE] Add cache support for GEM's admin bucket. The cache will be enabled by default when you use the
  small.yaml, large.yaml, capped-small.yaml or capped-large.yaml Helm values file. #3740
  > **Note:** For more information, refer to the [Grafana Enterprise Metrics configuration](https://grafana.com/docs/enterprise-metrics/latest/config).
```

## 4.2.0

> * [FEATURE] Add cache support for GEM's admin bucket. #3740

Reviewer: Should go to main/unreleased

Reviewer: Also since the changelog is the primary source of information for upgrade, let's mention that it will be enabled by default in the small, large values files.

Author: > Should go to main/unreleased
Sorry, I keep forgotten whether to put changelog in the upcoming release or in main/unreleased. 

> Also since the changelog is the primary source of information for upgrade, let's mention that it will be enabled by default in the small, large values files.
Added the comment

## 4.1.0

* [CHANGE] Configured `max_total_query_length: 12000h` limit to match Mimir jsonnet-based deployment. #3879

> * [FEATURE] Add cache support for GEM's admin bucket. #3740

Reviewer 2: Can you change the place of this changelog entry please? This is in the 4.1.0 section which was already released

Reviewer 2: also nit: if you have a link to the GEM docs for this feature and/or if you can add why a user might enable this feature. That would make it slightly easier for users to know if this feature concerns them.

> * [FEATURE] Add cache support for admin bucket. #3740

Reviewer 2: we usually order the changelog like

```
- [CHANGE]
- [FEATURE]
- [ENHANCEMENT]
- [BUGFIX]
```

can you please move this 3 lines up so it's at the top?

Also, to be super clear you can mention that this change only applies to GEM
```suggestion
* [FEATURE] Add cache support for GEM's admin bucket. #3740
```

Author: Fixed
